apps/courses/cms_toolbars.py

```diff
@@ -0,0 +1,57 @@
+"""
+Toolbar extension for the courses application
+"""
+from django.utils.translation import ugettext_lazy as _
+
+from cms.api import get_page_draft
+from cms.extensions.toolbar import ExtensionToolbar
+from cms.toolbar_pool import toolbar_pool
+from cms.utils.page_permissions import user_can_change_page
+
+from .models import Course
+
+
+class BaseExtensionToolbar(ExtensionToolbar):
+    """
+    This extension class customizes the toolbar for our page extensions.
+
+    It must be inherited to define the `model` class attribute.
+    """
+
+    def populate(self):
+        """
+        When visiting a page that is linked to a page extension, show an item in the toolbar only
+        to users who have the permission to update the page.
+
+        Clicking on this menu item will open the admin page of the page extension.
+
+        Note: We don't want to show a link to create a new page extension on a page that does not
+        have it yet. For clariry, we choose to only create page extensions transparently via the
+        page creation wizard.
+        """
+        # always use draft if we have a page
+        self.page = get_page_draft(self.request.current_page)
+        if not self.page:
+            # Nothing to do
+            return
+
+        if user_can_change_page(user=self.request.user, page=self.page):
+            page_extension, admin_url = self.get_page_extension_admin()
+            if page_extension:
+                edit_mode_inactive = not self.toolbar.edit_mode_active
+                page_menu = self.toolbar.get_or_create_menu("page")
+                # Create the new menu item as a modal
+                page_menu.add_modal_item(
+                    _("{:s} settings").format(self.model.__name__),
+                    url=admin_url,
+                    disabled=edit_mode_inactive,
+                    position=4,
```

Comment:
Glad you ask ;-)

I tried to get the position of the "Page settings..." item to put it below but it seems that the toolbar is built in now predefined order so sometimes it is not there yet. There's a way to order alphabetically but that's not what we want... etc. 

So. A better way than what I did is maybe to have a setting... Maybe we can have a shared setting for all page extensions since they can not coexist in our case. 

Should I call it PAGE_EXTENSION_TOOLBAR_ITEM_POSITION ?

Reply:
Yes! Should we prefix this kind of setting with `RICHIE_` to quickly identify that it's a custom settings?

Reply:
Maybe we should have a `settings.py` file in the app and the RICHIE_ prefix is only required to override the setting in the project conf?

Reply:
:thinking: I'm not sure this is a good idea to dilute configuration in multiple files.

Reply:
I have in mind to later package "courses" and "search" as third party applications so that people can start their own CMS project and add these applications as functionalities to their CMS. 

That's why I think it's good that defaults for our settings are handled in the application itself.

```python
from django.conf import settings

PAGE_EXTENSION_TOOLBAR_ITEM_POSITION = getattr(
    settings, "RICHIE_PAGE_EXTENSION_TOOLBAR_ITEM_POSITION", 4)
```

Reply:
Ok, I wasn't aware that we might package those apps separately. In this case, that makes sense to me.

Instead of naming this file `courses/settings.py` shouldn't we call it `courses/defaults.py`? I'm not sure if this is a convention/good practice, but this is how I usually proceed.

Reply:
Let's do that